lib/services/access_request.go

```diff
@@ -859,13 +895,32 @@ func (c *ReviewPermissionChecker) CanReviewRequest(req types.AccessRequest) (boo
 	needsAllow := make([]string, len(requestedRoles))
 	copy(needsAllow, requestedRoles)
 
+	// Users are allowed to promote a request if the user can grant all of the roles requested
+	canPromote := false
+	for _, grantedRoles := range c.AccessListsToGrants {
+		needsGrant := make([]string, len(requestedRoles))
+		copy(needsGrant, requestedRoles)
+		for _, role := range needsGrant {
+			if _, ok := grantedRoles[role]; ok {
+				// Trim off the matched needs grant.
+				needsGrant = needsGrant[1:]
+			}
+		}
+
+		// If there are no other needsGrant roles left, this user can review/promote.
+		if len(needsGrant) == 0 {
+			canPromote = true
+			break
+		}
+	}
```

Comment:
Promotion only works for resource access requests currently so this will not do anything I'm pretty sure.

Reply:
This does function today, actually, but I agree with the sentiment. I was hesitant about pushing up the PR in general because the UX isn't 100% confusing. I'll close this for now and we can take another long another time.